features/step_definitions/controller_node/background_steps.rb

```diff
@@ -15,5 +15,11 @@
 end
 
 Given(/^the controller node is in "([^"]*)" state$/) do |ready|
+  # before checking for "ready" state, make sure the controller is not
+  # in an "unknown" state
+  20.times do
+    break unless control_node.status == "unknown"
+    sleep 5
```

Comment:
Add a if conditional with a time out here.

``if time > [20|40|60]:
    break # Or something similar.
end``

Additionally these conditional timeouts should be variables for reuse if required.